docs/concepts/macros/sqlmesh_macros.md

```diff
@@ -498,6 +498,11 @@ SELECT
 FROM table
 ```
 
+!!! note
+
+    The `@IF` macro does not short-circuit based on the condition's value; its arguments are always rendered. This
```

Comment:
Hmm...this is a tough one - might be too much to fit into a note. Wdyt?

Possible text:

SQLMesh macro arguments can contain placeholder values, such as local variables. 

Before a macro can evaluated, SQLMesh must substitute the actual values into these placeholders - that process is called "rendering." Rendering is always the first step in SQLMesh's macro processing.

Rendering occurs **before** the `@IF` condition is evaluated. For example, SQLMesh cannot determine whether `my_column > @my_value` until it has first substituted the number `@my_value` represents.

Your macro might do things besides returning a value, such as printing a message or executing a statement (i.e., the macro "has side effects"). The side effect code will always run during the rendering step. To prevent, modify the macro code to condition the side effects on the evaluation stage.

Reply:
Fair enough, I think I over-condensed it. I'd keep it shorter, though. Don't we talk about rendering elsewhere within the SQLMesh macros page? I think I'd keep the last two paragraphs in your suggestion.

Reply:
Good call on the last two paragraphs. How about this?

----------

[Macro rendering](#sqlmesh-macro-approach) occurs before the `@IF` condition is evaluated. For example, SQLMesh cannot determine whether `my_column > @my_value` until it has first substituted the number `@my_value` represents.

Your macro might do things besides returning a value, such as printing a message or executing a statement (i.e., the macro "has side effects"). The side effect code will always run during the rendering step. To prevent this, modify the macro code to condition the side effects on the evaluation stage.